src/js/base/runtime.js

```diff
@@ -6260,6 +6260,8 @@ function (Namespace, jsnums, codePoint, util, exnStackParser, loader, seedrandom
       },
       'clearParam' : clearParam,
       'stdout' : theOutsideWorld.stdout,
+      'stdout_fd' : theOutsideWorld.stdout_fd,
```

Comment:
I'm not thrilled by the redundancy of `stdout` and `stdout_fd` -- why do you need the file descriptors?  (Surely that makes things harder to mock/emulate, not easier...)  Also, see below about async vs sync stuff -- you might be able to implement this more easily with promises, `readline` and the normal `stdin`.

Reply:
It did make things harder - fixed with the newer implementation.